src/System.Threading.Tasks/tests/System.Runtime.CompilerServices/AsyncTaskMethodBuilderTests.cs

```diff
@@ -548,7 +548,15 @@ async Task YieldOnceAsync(object s)
                 }
 
                 var state = new InvokeActionOnFinalization { Action = () => tcs.SetResult(true) };
-                var al = new AsyncLocal<object> { Value = state }; // ensure the object is stored in ExecutionContext
+                var al = new AsyncLocal<object>(args => {
+                    // Temporary logging to get more info when the test timeout to look who hold a reference to the finalizer object.
+                    string currentValue = args.CurrentValue == null ? "'null'" : "'Object'";
+                    string previousValue = args.PreviousValue == null ? "'null'" : "'Object'";
+                    Console.WriteLine($"AsyncMethodsDropsStateMachineAndExecutionContextUponCompletion: Thread Id: {Thread.CurrentThread.ManagedThreadId} Current Value: {currentValue}  Previous Value: {previousValue} ThreadContextChanged: {args.ThreadContextChanged}");
```

Comment:
This is going to write to the console regardless of whether the test passes or fails.  We try hard to avoid doing that.